### PR TITLE
Add detection exploit for CVE-2024-43044

### PR DESCRIPTION
**Pull request by ybadaoui-ostorlab**

### CODE REVIEW

File: agent/exploits/cve_2024_43044.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from urllib import parse as urlparse
+
+from agent.exploits import webexploit
+from agent import exploits_registry
+from agent import definitions
+
+
+VULNERABILITY_TITLE = "JENKINS ARBITRARY FILE READ TO RCE"
+VULNERABILITY_REFERENCE = "CVE-2024-43044"
+VULNERABILITY_DESCRIPTION = """Arbitrary file read vulnerability through agent connections can lead to RCE
```
Comment (deadly-panda) [inline agent/exploits/cve_2024_43044.py:16]:
```suggestion
VULNERABILITY_DESCRIPTION = """Arbitrary file read vulnerability through agent connections can lead to RCE.
```
Also, I think you need a bit more details..

File: agent/exploits/cve_2024_43044.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        try:
+            req = requests.Request(
+                method=self.check_request.method,
+                url=target_endpoint,
+                data=self.check_request.data,
+            ).prepare()
+            if self.check_request.headers is not None:
+                self.check_request.headers.update(req.headers)
+                req.headers = self.check_request.headers  # type: ignore
+            resp = session.send(req, timeout=DEFAULT_TIMEOUT, verify=False)
```
Comment (deadly-panda) [inline agent/exploits/cve_2024_43044.py:65]: do u need `verify=False` if u already set it on line 50?

Comment (PiranhaSa) [reply agent/exploits/cve_2024_43044.py:65]: no need

Comment (ybadaoui-ostorlab) [reply agent/exploits/cve_2024_43044.py:65]: this one is from the base class webexploit i just override the version check

**Review by nmasdoufi-ol — APPROVED** [review]

File: agent/exploits/cve_2024_43044.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+from agent import exploits_registry
+from agent import definitions
+
+
+VULNERABILITY_TITLE = "JENKINS ARBITRARY FILE READ TO RCE"
+VULNERABILITY_REFERENCE = "CVE-2024-43044"
+VULNERABILITY_DESCRIPTION = """Arbitrary file read vulnerability through agent connections can lead to RCE.
+"""
+RISK_RATING = "CRITICAL"
+# todo update the version to test also for LTS
```
Comment (ErebusZ) [inline agent/exploits/cve_2024_43044.py:19]:
```suggestion
# TODO(youssef): update the version to test also for LTS
```

**Review by ErebusZ — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**